### PR TITLE
Configure with Reaper with Cassandra role

### LINKED ISSUE #153: Generate a Cassandra user for Reaper
**Is your feature request related to a problem? Please describe.**
If authentication is enabled (see #151), we should create a Cassandra user for Reaper.

**Describe the solution you'd like**
There are a few things that need to be done:

* Generate C* credentials
* Create a secret that stores the credentials
* Specify the Reaper user and secret in the cassdc.yaml template
* Specify the secret in the reaper.yaml template 
  * https://github.com/k8ssandra/reaper-operator/issues/28 needs to be done first

### PR DESCRIPTION
**Pull request by jsanda**
Fixes #153.

#208 needs to be merged before this.

### CODE REVIEW

**Comment by jsanda** [comment]
The following info is relevant for testing purposes.

See the comments [here](https://github.com/jsanda/k8ssandra/blob/reaper-user/charts/k8ssandra/values.yaml#L56) of the new properties.

To obtain the credentials you can use the following:

```
$ kubectl get secret <secret name> -o json | jq -r '.data.username' | base64 --decode

$ kubectl get secret <secret name> -o json | jq -r '.data.password' | base64 --decode
```

I suggest then verifying those credentials can be used to access cqlsh one on the of C* pods as follows:

```
$ kubectl exec -it <cassandra-pod> -c cassandra -u <username> -p <password>
```

File: tests/unit/template_cassdc_test.go [hunk 1]
```diff
@@ -293,6 +296,46 @@ var _ = Describe("Verify CassandraDatacenter template", func() {
 			Expect(cassdc.Spec.SuperuserSecretName).To(Equal(clusterName + "-superuser"))
 		})
 
+		//It("enabling Cassandra auth", func() {
+		//	dcName := "test"
+		//	clusterSize := 3
+		//
+		//	authCachePeriod := int64(7200000)
+		//	cacheValidityPeriod := authCachePeriod + 1
+		//	cacheUpdateInterval := authCachePeriod + 2
```
Comment (burmanm) [inline tests/unit/template_cassdc_test.go:305]: This commented out code should be probably removed?

**Comment by jsanda** [comment]
@jakerobb and/or @jeffbanks can you please review this?

**Review by jeffbanks — APPROVED** [review]
LGTM

**Comment by jsanda** [comment]
@jeffbanks Can you give another quick review? I had to bump the chart version.

**Pull request merged**